akka-cluster-sharding/src/multi-jvm/scala/akka/cluster/sharding/MultiNodeClusterShardingSpec.scala

```diff
@@ -0,0 +1,134 @@
+    }
+
+}
+
+abstract class MultiNodeClusterShardingSpec(val config: MultiNodeClusterShardingConfig)
+    extends MultiNodeSpec(config)
+    with MultiNodeClusterSpec {
+
+  import MultiNodeClusterShardingSpec._
+  import config._
+
+  override def initialParticipants: Int = roles.size
+
+  protected val storageLocations = List(
+    new File(system.settings.config.getString("akka.cluster.sharding.distributed-data.durable.lmdb.dir")).getParentFile)
+
+  override protected def atStartup(): Unit = {
+    storageLocations.foreach(dir => if (dir.exists) FileUtils.deleteQuietly(dir))
+    enterBarrier("startup")
+    super.atStartup()
+  }
+
+  override protected def afterTermination(): Unit = {
+    storageLocations.foreach(dir => if (dir.exists) FileUtils.deleteQuietly(dir))
+    super.afterTermination()
+  }
+
+  protected def join(from: RoleName, to: RoleName): Unit = {
+    runOn(from) {
+      Cluster(system).join(node(to).address)
+      awaitAssert {
+        Cluster(system).state.isMemberUp(node(from).address)
+      }
+    }
+    enterBarrier(from.name + "-joined")
+  }
+
+  protected def startSharding(
+      sys: ActorSystem,
+      entityProps: Props,
+      dataType: String,
+      extractEntityId: ShardRegion.ExtractEntityId = defaultExtractEntityId,
+      extractShardId: ShardRegion.ExtractShardId = defaultExtractShardId,
+      handOffStopMessage: Any = PoisonPill): ActorRef = {
+
+    ClusterSharding(sys).start(
+      typeName = dataType,
+      entityProps = entityProps,
+      settings = ClusterShardingSettings(sys).withRememberEntities(rememberEntities),
+      extractEntityId = extractEntityId,
+      extractShardId = extractShardId,
+      ClusterSharding(sys).defaultShardAllocationStrategy(ClusterShardingSettings(sys)),
+      handOffStopMessage)
+  }
+
+  protected def isDdataMode: Boolean = mode == ClusterShardingSettings.StateStoreModeDData
+
+  private def setStoreIfNotDdataMode(sys: ActorSystem, storeOn: RoleName): Unit =
+    if (!isDdataMode) {
+      val probe = TestProbe()(sys)
+      sys.actorSelection(node(storeOn) / "user" / "store").tell(Identify(None), probe.ref)
+      val sharedStore = probe.expectMsgType[ActorIdentity](20.seconds).ref.get
+      SharedLeveldbJournal.setStore(sharedStore, sys)
+    }
+
+  /**
+   * {{{
+   *    startPersistence(startOn = first, setStoreOn = Seq(first, second, third))
+   * }}}
+   *
+   * @param startOn the node to start the `SharedLeveldbStore` store on
+   * @param setStoreOn the nodes to `SharedLeveldbJournal.setStore` on
+   */
+  protected def startPersistenceIfNotDdataMode(startOn: RoleName, setStoreOn: Seq[RoleName]): Unit =
+    if (!isDdataMode) {
+
+      Persistence(system)
+      runOn(startOn) {
+        system.actorOf(Props[SharedLeveldbStore], "store")
+      }
+      enterBarrier("persistence-started")
+
+      runOn(setStoreOn: _*) {
+        setStoreIfNotDdataMode(system, startOn)
+      }
+
+      enterBarrier(s"after-${startOn.name}")
+
+    }
+
+}
```

Comment:
And this, similar in quite a few multi tests.

Reply:
Do you plan to use it from all those tests in this PR as well (if not make sure to ticket that)

Reply:
Created https://github.com/akka/akka/issues/27749 - can do on the plane ✈️